### ISSUE TITLE
DBEntity controller -> convert_to_sa needs to be able to deal with SQLAlchemy synonyms

### ISSUE THREAD

**Issue by kbower** [kbower/tickettest1#213]
_Imported from trac ticket 213.
Created by **rneville**
Opened in trac: May 21, 2010, 9:50:58 AM
Last modified in trac: May 25, 2010, 9:34:53 AM_
***
_by rneville:_
see auth.py...

```
 password = synonym('_password', descriptor=property(_get_password,
                                                        _set_password))
```
when saving a transferobject, this pukes on line 287 in the dbentity controller, complaining that SynonymProperty object has not attribute 'argument'.

Output from debug view:
```
attribs	{'_password': <sqlalchemy.orm.properties.ColumnProperty object at 0x2b8bc5473150>, 'created': ...
dict_obj	{u'created': None, u'display_name': u'Test User', u'email_address': u'test@somewhere.com', u'groups' ...
field	u'password'
field_class	<class 'pylotengine.model.auth.Group'>
field_prop	<sqlalchemy.orm.properties.SynonymProperty object at 0x2b8bc5442d10>
p	<sqlalchemy.orm.properties.RelationshipProperty object at 0x2b8bc59eb850>
sa_class	<class 'pylotengine.model.auth.User'>
sa_obj	<User: email="None", display name="Test User">
self	<pylotengine.controllers.user.UserController object at 0x2b8bc5a9c2d0>
type_of_column	<class 'sqlalchemy.orm.properties.SynonymProperty'>
value	u'test'
```
code fragment in question:

```python
else:
                           # assume type_of_column == RelationProperty for now
                           if type(field_prop.argument) == Mapper:
                               # I think this means this is a backref
                               field_class = field_prop.argument.class_
```

**Comment by kbower** [comment]
_**kbower** trac comment on May 25, 2010, 9:34:53 AM:_

(In [[537]](../commit/69cf4792701a82ec739116dc1e72a9756a5c138d)) Allow SynonymProperty to function on a save service call.  Fixes #213

**Comment by kbower** [comment]
_**kbower** trac status change on May 25, 2010, 9:34:53 AM:_ **closed**
